### PR TITLE
Forum post standalone page

### PR DESCRIPTION
**Pull request by Devorein**
[Task Card](https://app.charmverse.io/charmverse/page-49366552253645923?viewId=d15c9b90-8918-44da-bbd8-db8712faa723&cardId=876aa6ac-5a5f-4262-b5f9-730052f1a534)

### CODE REVIEW

File: components/forum/components/PostDialog/PostDialog.tsx [hunk 1]
```diff
@@ -75,19 +78,21 @@ export default function PostDialog(props: Props) {
...
-          //     </Button>
-          //   </Box>
-          // }
+          toolbar={
+            page && (
+              <Box display='flex' justifyContent='space-between'>
+                <Button
+                  size='small'
+                  color='secondary'
+                  href={`/${router.query.domain}/forum/post/${page.id}`}
```
Comment (mattcasey) [inline components/forum/components/PostDialog/PostDialog.tsx:87]: should be page.path

File: components/forum/components/PostPage/PostPage.tsx [hunk 2]
```diff
@@ -148,21 +149,23 @@ export function PostPage({ page, spaceId, onSave }: Props) {
...
-          isContentControlled={true}
-          content={form.content as PageContent}
-          onContentChange={updatePostContent}
-        >
-          <PageTitleInput readOnly={readOnly} value={form.title} onChange={updateTitle} />
-          <Box my={2}>
-            <PostCategoryInput spaceId={spaceId} setCategoryId={updateCategoryId} categoryId={categoryId} />
-          </Box>
-        </CharmEditor>
+        {page && user && (
```
Comment (mattcasey) [inline components/forum/components/PostPage/PostPage.tsx:152]: why did you have to add page && user here? we are not using 'user' in the props, and pageId is optional

File: pages/[domain]/forum/post/[postId].tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import useSWR from 'swr';
+
+import charmClient from 'charmClient';
+import ErrorPage from 'components/common/errors/ErrorPage';
+import getPageLayout from 'components/common/PageLayout/getLayout';
+import { PostPage } from 'components/forum/components/PostPage/PostPage';
+
+export default function BlocksEditorPage() {
+  const router = useRouter();
+  const pagePath = router.query.postId as string;
```
Comment (mattcasey) [inline pages/[domain]/forum/post/[postId].tsx:11]: rename to router.query.path or router.query.pagePath

File: pages/[domain]/forum/post/[postId].tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { useRouter } from 'next/router';
```
Comment (mattcasey) [inline pages/[domain]/forum/post/[postId].tsx:1]: Please use page path instead for navigation and rename this file to [pagePath].tsx

**Review by mattcasey — CHANGES_REQUESTED** [review]
- We should set the page title using `hooks/usePageTitle`.
- Breadcrumbs should look like: `Forum > [page title]` (with Forum linking back to main Forum feed). It would be nice to include the category: `Forum > [category title] > [page title]`. I think this is where having category in the path would be nice, because you could just consume useForumCategories() and match on id/path.

File: components/forum/components/PostPage/PostPage.tsx (hunk 2 above)
Comment (Devorein) [reply components/forum/components/PostPage/PostPage.tsx:152]: The readOnly property is false initially as `user` is `null`. Charmeditor doesn't pickup the updated readOnly property and so the editor becomes readOnly as well. I added a key of `user?.id` and it seems to be working now 🥳

**Pull request merged**